src/credentials.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+use std::ffi::CString;
+use std::ptr;
+
+use grpc_sys::{self, GrpcChannelCredentials, GrpcServerCredentials};
+use libc::c_char;
+
+pub struct ServerCredentialsBuilder {
+    root: Option<CString>,
+    cert_chains: Vec<*mut c_char>,
```

Comment:
can we use CString here, and get the raw ptr in build directly?

So we don't need to use into_raw, from_raw.